fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -703,12 +737,75 @@ private Collection<Long> distributionPrune(
         switch (distributionInfo.getType()) {
             case HASH: {
                 HashDistributionInfo info = (HashDistributionInfo) distributionInfo;
-                distributionPruner = new HashDistributionPruner(table.getTabletIdsInOrder(),
+                distributionKeys2TabletID.clear();
```

Comment:
clear all OlapScanNode prepared statement cached fields like  distributionKeys2TabletID in `lazyEvaluateRangeLocations`